homeworks/src/main/java/hw40/entity/Product.java

```diff
@@ -23,12 +23,13 @@ public class Product {
     @Column(name = "cost")
     private Double cost;
 
-    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY, mappedBy = "product")
-    private List<OrderProduct> orderProducts;
+    @ManyToOne(cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH}, fetch = FetchType.LAZY)
```

Comment:
The cascade type should be specified on the parent side only. Product is a child entity. Removing a product and an order associated with it is not correct.